src/DynamicDiscreteSamplers.jl

```diff
@@ -59,7 +59,7 @@ mutable struct SelectionSampler6
     p::NTuple{5, Float64}
     SelectionSampler6(p) = new(normalize(p))
 end
-function Base.rand(ss::SelectionSampler6)
+function Base.rand(rng::AbstractRNG, ss::SelectionSampler6)
     u = rand()
     count(<(u), ss.p) + 1
 end
```

Comment:
This doesn't actually use the `rng`. We should fix this and also add some tests that confirm that the passed RNG is the only one used. For example, checking that the global rng state is the same before and after sampling from a DiscreteDynamicSampler while passing an explicit rng.

Reply:
ah yes ops my bad